### PR TITLE
Add functions to search in periods including end time

### PR DESCRIPTION
**Pull request by greg-mclain**

### CODE REVIEW

**Comment by greg-mclain** [comment]
That's a good point. I briefly thought of implementing it that way yesterday but ultimately decided that this would be the quicker/less intrusive change, since we could restrict it to just the scenarios we're trying to fix. However, I think adding the property to floating period would be better in the long term (assuming we never have the scenario you mentioned in which the same period is used for an end time inclusive search in one case and exclusive in another). I'll go that route.

File: periodic_test.go [hunk 1]
```diff
@@ -109,57 +109,57 @@ func TestPeriod_Contains(t *testing.T) {
...
-			NewPeriod(testTime1, testTime2.Add(-time.Minute)),
+			NewPeriod(time.Time{}, testTime2, false),
+			NewPeriod(testTime1, testTime2.Add(-time.Minute), false),
 		}, {
 			"True when period has no start and no end",
 			true,
-			NewPeriod(time.Time{}, time.Time{}),
-			NewPeriod(testTime1, testTime2),
+			NewPeriod(time.Time{}, time.Time{}, false),
+			NewPeriod(testTime1, testTime2, false),
```
Comment (briancharous) [inline periodic_test.go]: Add some tests for containment of periods with inclusive end times

File: periodic_test.go [hunk 2]
```diff
@@ -518,27 +538,27 @@ func TestApplicableDays_AnyApplicable(t *testing.T) {
 func TestPeriod_Equals(t *testing.T) {
```
Comment (briancharous) [inline periodic_test.go]: Should 2 periods be equal if one is inclusive on the end and one isn't? I think the only place `Equals` is used is on the `Update` method on the tree. It checks if the period currently stored in the tree is equal to the new period for that key, and if they are, it just swaps the contents of the node; if the period has changed, it deletes the node and inserts a new one. Swapping contents is obviously a lot better performance wise. But there's the edge case where the period hasn't changed time-wise but its inclusivity has. Technically I don't think the node needs to be deleted and re-inserted, but the inclusivity property on the period stored in the node has to change.

What I'm getting at here is that I think that `Equals` on `Period` needs to also check if `EndsInclusive` is the same. That way `Update` can remain unchanged. There might be a tiny performance hit on the tree but it's such an edge case that I think it makes sense to update `Equals` to be logically consistent.

File: period_collection.go [hunk 3]
```diff
@@ -389,7 +389,7 @@ func (pc *PeriodCollection) containsTime(root *node, time time.Time) bool {
 	if root.leaf {
 		return false
 	}
-	if root.period.ContainsTime(time) {
+	if root.period.ContainsTime(time, false) {
```
Comment (briancharous) [inline period_collection.go]: Fine for now since it preserves the behavior but we might want to make this a flag passable to `ContainsTime` in the future

**Pull request merged**